ctapipe/irf/select.py

```diff
@@ -0,0 +1,205 @@
+            n_showers=show["n_entries"].sum(),
+            energy_min=sim["energy_range_min"].quantity[0],
+            energy_max=sim["energy_range_max"].quantity[0],
+            max_impact=sim["max_scatter_range"].quantity[0],
+            spectral_index=sim["spectral_index"][0],
+            viewcone_max=sim["max_viewcone_radius"].quantity[0],
+            viewcone_min=sim["min_viewcone_radius"].quantity[0],
+        )
+
+        return (
+            sim_info,
+            PowerLaw.from_simulation(sim_info, obstime=obs_time),
+            obs,
+        )
+
+    def make_derived_columns(self, events, spectrum, obs_conf, fov_bins):
+        if obs_conf["subarray_pointing_lat"].std() < 1e-3:
+            assert all(obs_conf["subarray_pointing_frame"] == 0)
+            # Lets suppose 0 means ALTAZ
+            events["pointing_alt"] = obs_conf["subarray_pointing_lat"][0] * u.deg
+            events["pointing_az"] = obs_conf["subarray_pointing_lon"][0] * u.deg
+        else:
+            raise NotImplementedError(
+                "No support for making irfs from varying pointings yet"
+            )
+
+        events["theta"] = calculate_theta(
+            events,
+            assumed_source_az=events["true_az"],
+            assumed_source_alt=events["true_alt"],
+        )
+        events["true_source_fov_offset"] = calculate_source_fov_offset(
+            events, prefix="true"
+        )
+        events["reco_source_fov_offset"] = calculate_source_fov_offset(
+            events, prefix="reco"
+        )
+        # TODO: Honestly not sure why this integral is needed, nor what
+        # are correct bounds
+        if self.kind == "gamma":
+            spectrum = spectrum.integrate_cone(
+                fov_bins.fov_offset_min * u.deg, fov_bins.fov_offset_max * u.deg
+            )
+        self.log.info("kind: %s" % self.kind)
+        self.log.info("target: %s" % self.target_spectrum)
+        self.log.info("spectrum: %s" % spectrum)
+        events["weight"] = calculate_event_weights(
+            events["true_energy"],
+            target_spectrum=self.target_spectrum,
+            simulated_spectrum=spectrum,
+        )
+
+        return events
+
+
+class EventPreProcessor(QualityQuery):
+    """Defines preselection cuts and the necessary renaming of columns"""
+
+    energy_reconstructor = Unicode(
+        default_value="RandomForestRegressor",
+        help="Prefix of the reco `_energy` column",
+    ).tag(config=True)
+    geometry_reconstructor = Unicode(
+        default_value="HillasReconstructor",
+        help="Prefix of the `_alt` and `_az` reco geometry columns",
+    ).tag(config=True)
+    gammaness_classifier = Unicode(
+        default_value="RandomForestClassifier",
+        help="Prefix of the classifier `_prediction` column",
+    ).tag(config=True)
+
+    quality_criteria = List(
+        default_value=[
+            ("multiplicity 4", "np.count_nonzero(tels_with_trigger,axis=1) >= 4"),
```

Comment:
Do we really want a multiplicity cut of 4 by default? I would remove that and only require the valid reconstruction.